solutions/minwoo/codetree/나무박멸.py

```diff
@@ -0,0 +1,105 @@
+n,m,k,c = map(int,input().split())
+board = [list(map(int,input().split())) for _ in range(n)]
+kill = [[0]*n for _ in range(n)]
+ans = 0
+
+def step1(): # 나무성장
+    for i in range(n):
+        for j in range(n):
+            adj_tree = 0
+            if board[i][j] > 0 :
+                for dy,dx in ((0,-1),(0,1),(1,0),(-1,0)):
+                    ny = i+dy
+                    nx = j+dx
+                    if 0<=ny<n and 0<=nx<n and board[ny][nx] > 0 :
+                        adj_tree +=1
+                board[i][j] += adj_tree
+
+def step2(): # 번식
+    new_board = [[0]*n for _ in range(n)] # 번식 판
+    for i in range(n):
+        for j in range(n):
+            adj_tree = 0
+            if board[i][j] > 0:
+                for dy, dx in ((0, -1), (0, 1), (1, 0), (-1, 0)):
+                    ny = i + dy
+                    nx = j + dx
+                    if 0<=ny<n and 0<=nx<n and board[ny][nx] == 0 and kill[ny][nx] == 0: # 번식이 가능한칸을 카운트
+                        adj_tree +=1
+
+                if adj_tree == 0 : # zero devide 방어코드
+                    continue
+                for dy, dx in ((0, -1), (0, 1), (1, 0), (-1, 0)): # 번식을 새로운 판에 진행.
+                    ny = i+dy
+                    nx = j+dx
+                    if 0<=ny<n and 0<=nx<n and board[ny][nx] == 0 and kill[ny][nx] == 0:
+                        new_board[ny][nx] += (board[i][j] // adj_tree)
```

Comment:
```suggestion
            if board[i][j] <= 0: continue
            for dy, dx in ((0, -1), (0, 1), (1, 0), (-1, 0)):
                ny = i + dy
                nx = j + dx
                if 0<=ny<n and 0<=nx<n and board[ny][nx] == 0 and kill[ny][nx] == 0: # 번식이 가능한칸을 카운트
                    adj_tree +=1

            if adj_tree == 0 : # zero devide 방어코드
                continue
            for dy, dx in ((0, -1), (0, 1), (1, 0), (-1, 0)): # 번식을 새로운 판에 진행.
                ny = i+dy
                nx = j+dx
                if 0<=ny<n and 0<=nx<n and board[ny][nx] == 0 and kill[ny][nx] == 0:
                    new_board[ny][nx] += (board[i][j] // adj_tree)
```

1. 조금 사소한 부분 같지만, 이 부분에서도 indent를 줄이는 방식으로 코드를 써보는 게 어떨까요 ㅎㅎ 베어에 있을 때 준서님께 리뷰 요청 드리면 매번 꼼꼼히 지적해주셨던 부분이라 민우님께도 전달드립니다..ㅋㅋㅋㅋ
2. 위의 for문에서는 adj_tree의 값을 계산하고 뒤의 for문에서는 new_board의 값을 업데이트하고 있다는 점을 제외하면 두 for문은 거의 똑같은데, 이처럼 코드가 똑같이 반복되고 있는 것을 줄이는 로직을 고민해보시길 추천드립니다! 이 정도 간단한 반복은 큰 문제 없지 않나 할 수 있긴 한데, 제 경험상 되도록이면 반복은 줄일수록 좋더라고요. 디버깅 하다 보면 뒤의 for문에서 문제가 있다는 걸 일찍 발견해서 2시간 디버깅했는데 안고쳐져서 계속 보다보니 그 위에 있는 for문도 같이 고쳤어야 했던 경우가 너무 많아서 ㅠㅠ 제가 이 문제 풀어봤을 때도 그랬고요.